agenda/data/src/main/java/com/gkp/agenda/data/OfflineAgendaRepository.kt

```diff
@@ -0,0 +1,35 @@
+package com.gkp.agenda.data
+
+import android.annotation.SuppressLint
+import com.gkp.agenda.domain.AgendaRepository
+import com.gkp.auth.domain.session.SessionStorage
+import com.gkp.core.domain.util.TaskyResult
+import com.gkp.core.network.TaskyRetrofitApi
+import com.gkp.core.network.util.networkApiCall
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+
+class OfflineAgendaRepository(
+    private val taskyRetrofitApi: TaskyRetrofitApi,
+    private val sessionStorage: SessionStorage,
+) : AgendaRepository {
+    private val scope = CoroutineScope(SupervisorJob())
```

Comment:
I would highly recommend to not create custom coroutine scopes in classes that don't really have a lifecycle (or might be bound to other lifecycles, like those of a ViewModel). Doing so completely moves away the control over the execution of suspend functions to this repository (I mean that other classes can no longer know when logging out finished in this case).

Instead, I would launch most coroutines in viewModelScope and expose suspend functions in such classes instead. Sometimes, if you want to make a suspending function explicitly non-cancellable, you can create an application bound scope (created in your App class) which you pass to the constructor here. But more often than not, you want to keep them cancellable.